### PR TITLE
[14.0][IMP] date_range: fully automatic generation from date range types

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Port of #347 

Replaces onchange methods by readwrite computed fields so that we can apply new() instead of odoo.tests.Form.

### CODE REVIEW

**Comment by Saran440** [comment]
Functional Test, LGTM  :+1:

**Review by Saran440 — APPROVED** [review]

**Comment by kittiu** [comment]
Thank you for this useful feature. Works for me too.

**Review by kittiu — APPROVED** [review]

**Comment by rrebollo** [comment]
Strange things are happening https://github.com/OCA/server-ux/issues/574.

File: date_range/wizard/date_range_generator.py [hunk 1]
```diff
@@ -70,14 +228,72 @@ def _compute_date_ranges(self):
...
+            self.date_start = last.date_end + relativedelta(days=1)
+        elif self.type_id.autogeneration_date_start:
+            self.date_start = self.type_id.autogeneration_date_start
+        else:  # default to the beginning of the current year
+            self.date_start = today.replace(day=1, month=1)
+
+    @api.depends("date_start")
+    def _compute_date_end(self):
+        if not self.type_id or not self.date_start:
+            return
```
Comment (alexis-via) [inline date_range/wizard/date_range_generator.py:276]: You can't write a compute method without looping on self. Here is the kind of crash you get when upgrading:
```
Traceback (most recent call last):
  File "/home/odoo/erp/odoo/odoo/service/server.py", line 1201, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/home/odoo/erp/odoo/odoo/modules/registry.py", line 89, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/odoo/erp/odoo/odoo/modules/loading.py", line 455, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/home/odoo/erp/odoo/odoo/modules/loading.py", line 347, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/home/odoo/erp/odoo/odoo/modules/loading.py", line 199, in load_module_graph
    registry.init_models(cr, model_names, {'module': package.name}, new_install)
  File "/home/odoo/erp/odoo/odoo/modules/registry.py", line 422, in init_models
    env['base'].flush()
  File "/home/odoo/erp/odoo/odoo/models.py", line 5497, in flush
    self.recompute()
  File "/home/odoo/erp/odoo/odoo/models.py", line 5956, in recompute
    process(field)
  File "/home/odoo/erp/odoo/odoo/models.py", line 5940, in process
    field.recompute(recs)
  File "/home/odoo/erp/odoo/odoo/fields.py", line 1177, in recompute
    self.compute_value(recs)
  File "/home/odoo/erp/odoo/odoo/fields.py", line 1199, in compute_value
    records._compute_field_value(self)
  File "/home/odoo/erp/odoo/odoo/models.py", line 4089, in _compute_field_value
    odoo.fields.determine(field.compute, self)
  File "/home/odoo/erp/odoo/odoo/fields.py", line 83, in determine
    return needle(*args)
  File "/home/odoo/erp/server-ux/date_range/wizard/date_range_generator.py", line 280, in _compute_date_end
    if not self.type_id or not self.date_start:
  File "/home/odoo/erp/odoo/odoo/fields.py", line 987, in __get__
    record.ensure_one()
  File "/home/odoo/erp/odoo/odoo/models.py", line 5050, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: date.range.generator(3, 4)
```

**Pull request merged**